module/PowerNSX.psm1

```diff
@@ -27225,7 +27230,41 @@ function New-NsxLoadBalancerMemberSpec {
         $xmlDoc.appendChild($xmlMember) | out-null
 
         Add-XmlElement -xmlRoot $xmlMember -xmlElementName "name" -xmlElementText $Name
-        Add-XmlElement -xmlRoot $xmlMember -xmlElementName "ipAddress" -xmlElementText $IpAddress
+        if ( $PSCmdlet.ParameterSetName -eq "ipaddress" ) {
+            Add-XmlElement -xmlRoot $xmlMember -xmlElementName "ipAddress" -xmlElementText $IpAddress
+        }
+        else {
+
+            if ($Member -is [System.Xml.XmlElement] ) {
+                $MemberMoref = $Member.objectId
+            }
+            elseif ( ($Member -is [string]) -and ($Member -match "^vm-\d+$|^resgroup-\d+$|^dvportgroup-\d+$" )) {
+                $MemberMoref = $Member
+            }
+            elseif ( ($Member -is [string] ) -and ( [guid]::tryparse(($Member -replace ".\d{3}$",""), [ref][guid]::Empty)) )  {
+                $MemberMoref = $Member
+            }
+            elseif (( $Member -is [string]) -and ( $NsxMemberTypes -contains ($Member -replace "-\d+$") ) ) {
+                $MemberMoref = $Member
+            }
+            elseif ( $Member -is [VMware.VimAutomation.ViCore.Interop.V1.VirtualDevice.NetworkAdapterInterop] ) {
+                #See NSX API guide 'Attach or Detach a Virtual Machine from a Logical Switch' for
+                #how to construct NIC id.
+                $vmUuid = ($Member.parent | get-view).config.instanceuuid
+                $MemberMoref = "$vmUuid.$($Member.id.substring($Member.id.length-3))"
+
+            }
+            elseif (( $Member -is [VMware.VimAutomation.ViCore.Interop.V1.VIObjectInterop]) -and ( $NsxMemberTypes -contains $Member.ExtensionData.MoRef.Type)) {
+                $MemberMoref = $Member.ExtensionData.MoRef.Value
+            }
+            else {
+                throw "Invalid member specified $($Member)"
+            }
+
+            #Create a new member node
+            Add-XmlElement -xmlRoot $xmlMember -xmlElementName "groupingObjectId" -xmlElementText $MemberMoref
```

Comment:
Add also a groupingObjectName because when you try to edit with GUI get some strange issue (display no name...)